File: backend/app/api/schemas/preferences.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PreferencesOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    preferred_contract_types: list[str] | None = None
    preferred_work_modes: list[str] | None = None
    preferred_locations: list[str] | None = None
    preferred_keywords: list[str] | None = None
    preferred_domains: list[str] | None = None
    exclude_keywords: list[str] | None = None
    minimum_duration_months: int | None = None
    created_at: datetime
    updated_at: datetime


class PreferencesUpdate(BaseModel):
    """Remplacement complet du profil (PUT sémantique).
    Un champ absent est remis à None — envoyer [] pour vider une liste."""

    preferred_contract_types: list[str] | None = None
    preferred_work_modes: list[str] | None = None
    preferred_locations: list[str] | None = None
    preferred_keywords: list[str] | None = None
    preferred_domains: list[str] | None = None
    exclude_keywords: list[str] | None = None
    minimum_duration_months: int | None = None

File: frontend/src/types/preferences.ts
export interface UserPreferences {
  id: string;
  preferred_contract_types: string[] | null;
  preferred_work_modes: string[] | null;
  preferred_locations: string[] | null;
  preferred_keywords: string[] | null;
  preferred_domains: string[] | null;
  exclude_keywords: string[] | null;
  minimum_duration_months: number | null;
  created_at: string;
  updated_at: string;
}

export interface PreferencesUpdate {
  preferred_contract_types?: string[] | null;
  preferred_work_modes?: string[] | null;
  preferred_locations?: string[] | null;
  preferred_keywords?: string[] | null;
  preferred_domains?: string[] | null;
  exclude_keywords?: string[] | null;
  minimum_duration_months?: number | null;
}
